app/keepers/keepers.go

```diff
@@ -396,38 +415,45 @@ func NewAppKeeper(
 	)
 	appKeepers.WasmKeeper = &wasmKeeper
 
-	icaHostStack := icahost.NewIBCModule(*appKeepers.ICAHostKeeper)
-	wasmStack := wasm.NewIBCHandler(appKeepers.WasmKeeper, appKeepers.IBCKeeper.ChannelKeeper, appKeepers.IBCKeeper.ChannelKeeper)
+	var icaHostStack ibctypes.IBCModule
+	icaHostStack = icahost.NewIBCModule(*appKeepers.ICAHostKeeper)
+	icaHostStack = ibcfee.NewIBCMiddleware(icaHostStack, *appKeepers.IBCFeeKeeper)
+
+	// RecvPacket, message that originates from core IBC and goes down to app, the flow is
+	// channel.RecvPacket -> fee.OnRecvPacket -> ibchooker.OnRecvPacket -> transfer.OnRecvPacket
+	transferStack := ibcfee.NewIBCMiddleware(appKeepers.IBCTransferHooksMiddleware, *appKeepers.IBCFeeKeeper)
+	icaControllerStack := ibcfee.NewIBCMiddleware(icaControllerIBCModule, *appKeepers.IBCFeeKeeper)
```

Comment:
we should delete this for pstake, cosmoshub won't allow new channel creation till there is fee middleware enabled on cosmos

Reply:
Done

Reply:
Done